### PR TITLE
Docs: Add instructions for rebasing a branch 

### PR DESCRIPTION
**Pull request by oskosk**
Adds a very basic and opinionated set of steps for rebasing a branch via command line.

#### Changes proposed in this Pull Request:

* Adds section to `git-workflow.md` 

#### Testing instructions:

Check the rendered version of this [new section](https://github.com/Automattic/jetpack/blob/add/rebase-instructions-for-a12s/docs/git-workflow.md#keeping-your-pr-up-to-date-if-you-have-commit-access-to-this-repo).

### CODE REVIEW

File: docs/git-workflow.md [hunk 1]
```diff
@@ -56,3 +56,24 @@ $ git rebase jetpack/master
 # Then push your changes to your forked version of Jetpack
 $ git push -f origin update/my-changes
 ```
+
+### Keeping your PR up to date if you have commit access to this repo
```
Comment (jeherve) [inline docs/git-workflow.md]: Aren't the instructions the same even if you don't have commit access?

**Review by jeherve — COMMENTED** [review]
I like it. I wonder, however, if we should be more detailed in our instructions? I had added the following link a bit higher on the page in that doc and it includes more info:
https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request

File: docs/git-workflow.md (hunk 1 above)
Comment (oskosk) [reply docs/git-workflow.md]: >Aren't the instructions the same even if you don't have commit access?

 no you need to add another remote reference (you have one for your fork of Jetpack, and you need another for the main Jetpack repo) .

Comment (oskosk) [reply docs/git-workflow.md]: About the link. I think it's healthy to have specific steps for our use case so we can directly link persons to these when reviewing PRs that get stalled or conflicting. Maybe we could improve the texts here.

Comment (jeherve) [reply docs/git-workflow.md]: Should we extend your info so it's valid for both folks with commit access and folks without, by mentioning where to fetch changes from?

Comment (oskosk) [reply docs/git-workflow.md]: I guess your intention is to unify and just leave one section. My intention was to have a clear set of pasteable commands for folks for which, just working with Git was a big step as a contributor.

Comment (jeherve) [reply docs/git-workflow.md]: That makes sense. I think we can have 2 sections. Maybe we can just extend the section you added to make it more detailed, i.e. check the remote you are using and fetch from the right one?

**Comment by haszari** [comment]
Happy to comment on this, as I recently had to do this, and am fairly unfamiliar with rebasing :)

It might be worth linking out to a good explanation of rebasing vs. merging, or "why rebase?". Here are some options: 

- https://www.atlassian.com/git/tutorials/merging-vs-rebasing
- https://git-scm.com/book/en/v2/Git-Branching-Rebasing
- https://www.atlassian.com/git/tutorials/rewriting-history/git-rebase

I'd comment the "goals" of these two lines separately, rather than combined:

```
# Update your fork's master branch with the latest changes from the original Jetpack repository
$ git fetch jetpack
# Rebase (reapply) your changes so your branch starts from the latest master
$ git rebase jetpack/master
```

Also I found that in my case, I could almost always use `git push --force-with-lease` rather than plain force, which is somewhat safer. 

Probably safer to use the full `--force` (or `-force-with-lease`) rather than simply `-f` in the examples; it should be clear that's what's happening!

**Comment by oskosk** [comment]
@haszari what's the safety that one gets by that ?

**Comment by haszari** [comment]
@oskosk Let me see if I can explain this  .. 

`--force` will obliterate history indiscriminately. If someone else has pushed commits (or even you, from a different clone!), they will get stomped.

`--force-with-lease` will only obliterate history in a particular scenario, and if the situation doesn't apply, it will fail safely instead. The expected scenario is that noone has pushed new commits to the remote branch; if someone else has pushed, `--force-with-lease` will back out, so you can pull down and re-rebase before pushing.

Adding some useful links as I'm not explaining this very well (I only recently learned about it, as I'm more familiar with a merge workflow).

http://weiqingtoh.github.io/force-with-lease/

https://developer.atlassian.com/blog/2015/04/force-with-lease/

**Pull request closed without merging**